src/dash-table/components/Table/props.ts

```diff
@@ -101,6 +119,27 @@ export interface IPaginationSettings {
     page_size: number;
 }
 
+export interface IUserInterfaceCell {
+    height: number;
+}
+
+export interface IUserInterfaceViewport {
+    scrollLeft: number;
+    scrollTop: number;
+    height: number;
+    width: number;
+}
+
+export interface IState {
+    forcedResizeOnly: boolean;
+    scrollbarWidth: number;
+    uiViewport?: IUserInterfaceViewport;
+    uiCell?: IUserInterfaceCell;
+    uiHeaders?: IUserInterfaceCell[];
```

Comment:
uiViewport - size and scroll offset of the viewport (.element `.row-1`)
uiCell - the height of a  row (assumed to be constant throughout the table)
uiHeaders - the height of the header rows